### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by praneetap**
#### Which issue(s) does this change fix?

Updating the readme 
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by hawflau — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,63 +1,58 @@
...
 
-[![Getting Started with AWS SAM](./media/get-started-youtube.png)](https://www.youtube.com/watch?v=1dzihtC5LJ0)
+The AWS Serverless Application Model (SAM) CLI is an open-source CLI tool that helps you develop serverless applications containing [Lambda functions](https://aws.amazon.com/lambda/), [Step Functions](https://aws.amazon.com/step-functions/), [Api Gateway](https://aws.amazon.com/api-gateway/), [EventBridge](https://aws.amazon.com/eventbridge/), [SQS](https://aws.amazon.com/sqs/), [SNS](https://aws.amazon.com/sns/) and more. Some of the features it provides are -
+- **Initialize serverless applications** in minutes with AWS provided infrastructure templates with `sam init`
+- **Compile, build, and package** Lambda functions with provided runtimes and with custom makefile workflows, for zip and image types of Lambda functions with `sam build`
+- **Locally test** a Lambda function and Api Gateway easily in a Docker container with `sam local` commands on SAM and CDK applications
+- **Sync and test your changes in the cloud** with `sam sync` in your developer environments
+- **Deploy** your SAM Tranform and Cloudformation templates using `sam deploy`
```
Comment (hoffa) [inline README.md:18]:
```suggestion
- **Deploy** your SAM and CloudFormation templates using `sam deploy`
```

File: README.md [hunk 2]
```diff
@@ -1,63 +1,58 @@
...
 ## What is this Github repository? 💻
 This Github Repository contains source code for SAM CLI. Here is the development team talking about this code:
 
 > SAM CLI code is written in Python. Source code is well documented, very modular, with 95% unit test coverage. 
 It uses this awesome Python library called Click to manage the command line interaction and uses Docker to run Lambda functions locally.
 We think you'll like the code base. Clone it and run `make pr` or `./Make -pr` on Windows!
 
+
+## Related Repositories and Resources
++ **SAM Transform** [Open source template specification](https://github.com/awslabs/serverless-application-model/) that provides shorthand syntax for Cloudformation
```
Comment (hoffa) [inline README.md:50]:
```suggestion
+ **SAM Transform** [Open source template specification](https://github.com/awslabs/serverless-application-model/) that provides shorthand syntax for CloudFormation
```

**Review by mndeveci — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,63 +1,58 @@
...
 
 ## Get Started
 
 To get started with building SAM-based applications, use the SAM CLI. SAM CLI provides a Lambda-like execution 
-environment that lets you locally build, test, debug, and deploy applications defined by SAM templates.
+environment that lets you locally build, test, debug, and deploy [AWS serverless](https://aws.amazon.com/serverless/) applications.
 
 * [Install SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
 * [Build & Deploy a "Hello World" Web App](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-quick-start.html)
 * [Install AWS Toolkit](https://aws.amazon.com/getting-started/tools-sdks/#IDE_and_IDE_Toolkits) to use SAM with your favorite IDEs.
```
Comment (hoffa) [inline README.md:30]:
```suggestion
* [Install AWS Toolkit](https://aws.amazon.com/getting-started/tools-sdks/#IDE_and_IDE_Toolkits) to use SAM with your favorite IDEs
```

File: README.md [hunk 4]
```diff
@@ -1,63 +1,58 @@
...
 This Github Repository contains source code for SAM CLI. Here is the development team talking about this code:
 
 > SAM CLI code is written in Python. Source code is well documented, very modular, with 95% unit test coverage. 
 It uses this awesome Python library called Click to manage the command line interaction and uses Docker to run Lambda functions locally.
 We think you'll like the code base. Clone it and run `make pr` or `./Make -pr` on Windows!
 
+
+## Related Repositories and Resources
++ **SAM Transform** [Open source template specification](https://github.com/awslabs/serverless-application-model/) that provides shorthand syntax for Cloudformation
++ **SAM CLI Application Init templates** Get started quickly with [predefined application templates](https://github.com/aws/aws-sam-cli-app-templates/blob/master/README.md) for all supported runtimes and languages, used by `sam init`
```
Comment (hoffa) [inline README.md:51]:
```suggestion
+ **SAM CLI application templates** Get started quickly with [predefined application templates](https://github.com/aws/aws-sam-cli-app-templates/blob/master/README.md) for all supported runtimes and languages, used by `sam init`
```

File: README.md [hunk 5]
```diff
@@ -1,63 +1,58 @@
...
++ **Lambda Builders** [Lambda builder tools](https://github.com/aws/aws-lambda-builders) for supported runtimes and custom build workflows, used by `sam build`
++ **Build and Local Emulation images for CI/CD tools** [Build container images](https://gallery.ecr.aws/sam/) to use with CI/CD tasks
```
Comment (hoffa) [inline README.md:53]:
```suggestion
+ **Build and local emulation images for CI/CD tools** [Build container images](https://gallery.ecr.aws/sam/) to use with CI/CD tasks 
```

**Review by hoffa — COMMENTED** [review]
Yay, great improvements!

Added a few suggestions.

**Comment by praneetap** [comment]
Thanks @hoffa! Updated.

**Review by hoffa — APPROVED** [review]
:shipit:

**Pull request merged**